src/administration/Administration.Service/BusinessLogic/ConnectorsBusinessLogic.cs

```
@@ -433,6 +435,19 @@ private async Task UpdateConnectorUrlInternal(Guid connectorId, ConnectorUpdateR
             con.ConnectorUrl = data.ConnectorUrl;
         });
 
+        if (connector.SelfDescriptionDocumentId != Guid.Empty)
+        {
+            documentRepository.AttachAndModifyDocument((Guid)connector.SelfDescriptionDocumentId!, null, doc =>
+            {
+                doc.DocumentStatusId = DocumentStatusId.INACTIVE;
+            });
+        }
+
+        var selfDescriptionDocumentUrl = $"{_settings.SelfDescriptionDocumentUrl}/{connector.SelfDescriptionCompanyDocumentId}";
```

Since the `connector.SelfDescriptionCompanyDocumentId` is nullable this should be checked... otherwise the callback from the sdFactory will always fail... If it is null you can throw a conflict exception

---

Added the check